qtwidgetsextra/groupbox/QAbstractButtonGroupBox.cpp

```diff
@@ -0,0 +1,341 @@
+#include "QAbstractButtonGroupBox.h"
+#include "QAbstractButtonGroupBox_p.h"
+
+#include <QGroupBox>
+#include <QButtonGroup>
+#include <QAbstractButton>
+#include <QBoxLayout>
+
+QAbstractButtonGroupBoxPrivate::QAbstractButtonGroupBoxPrivate(QAbstractButtonGroupBox *widgetP)
+    : QObject(widgetP)
+    , widget(widgetP)
+    , mainLayout(new QBoxLayout(QBoxLayout::TopToBottom, widget))
+    , group(new QButtonGroup(widget))
+    , orientation(Qt::Vertical)
+    , enabled(true)
+{
+    Q_ASSERT(widget);
+    connect(group,SIGNAL(buttonClicked(QAbstractButton*)),this,SLOT(buttonClicked(QAbstractButton*)));
+    connect(group,SIGNAL(buttonPressed(QAbstractButton*)),this,SLOT(buttonPressed(QAbstractButton*)));
+    connect(group,SIGNAL(buttonReleased(QAbstractButton*)),this,SLOT(buttonReleased(QAbstractButton*)));
+    connect(group,SIGNAL(buttonToggled(QAbstractButton*,bool)),this,SLOT(buttonToggled(QAbstractButton*,bool)));
+}
+
+void QAbstractButtonGroupBoxPrivate::clear() const
+{
+    QList<QAbstractButton*> buttons = group->buttons();
+    foreach (QAbstractButton *button, buttons) {
+        mainLayout->removeWidget(button);
+        group->removeButton(button);
+        delete button;
+    }
+}
+
+void QAbstractButtonGroupBoxPrivate::setOrientation(Qt::Orientation newOrientation)
```

Comment:
I'm not sure to understand the problem here. What function is duplicated? clear is in the Abstract class. the private one but in the public class it's a no sense.